manifests/server/config.pp

```diff
@@ -91,12 +91,48 @@
         quoted   => 'double',
       }
 
-      shellvar { 'slapd_sockets':
-        ensure   => present,
-        target   => '/etc/rc.conf',
-        variable => 'slapd_sockets',
-        value    => join($::openldap::server::ldapi_ifs, ' '),
-        quoted   => 'double',
+      if ($::openldap::server::ldapi_ifs) {
+        shellvar { 'slapd_sockets':
+          ensure   => present,
+          target   => '/etc/rc.conf',
+          variable => 'slapd_sockets',
+          value    => join($::openldap::server::ldapi_ifs, ' '),
+          quoted   => 'double',
+        }
+      }
+
+      if ($openldap::server::provider == 'olc') {
```

Comment:
This is getting quite complex. We might want to split this file by OS at some point.